### PR TITLE
Fix issue 317 - add SCS and SCC as legal target for IUT in TCT

### LINKED ISSUE #317: Inconsistency in linking for TCT and CLM
### Description

We have an inconsistency that should either be explained or changed:

* [CLM](https://github.com/eiffel-community/eiffel/blob/master/eiffel-vocabulary/EiffelConfidenceLevelModifiedEvent.md) allows to [SUBJECT](https://github.com/eiffel-community/eiffel/blob/master/eiffel-vocabulary/EiffelConfidenceLevelModifiedEvent.md#subject) link to CD, ArtC, as well as SCC and SCS 

* [TCT](https://github.com/eiffel-community/eiffel/blob/master/eiffel-vocabulary/EiffelTestCaseTriggeredEvent.md) link [Item Under Test (IUT)](https://github.com/eiffel-community/eiffel/blob/master/eiffel-vocabulary/EiffelTestCaseTriggeredEvent.md#iut) only allows CD or ArtC as targets, enforcing an extra [CD](https://github.com/eiffel-community/eiffel/blob/master/eiffel-vocabulary/EiffelCompositionDefinedEvent.md) with [ELEMENT](https://github.com/eiffel-community/eiffel/blob/master/eiffel-vocabulary/EiffelCompositionDefinedEvent.md#element) link to use SCC as IUT when running tests on source code.

### Motivation

Confusing why you can set a confidence level on for example a SCS event but you can't use the same event as a IUT link

### Exemplification

You can run test on source code and then put a CLM without having to introduce a CD

### Benefits

More logical protocol

### Possible Drawbacks

None that I can think of

### PR DESCRIPTION
**Pull request by m-linner-ericsson**
### Applicable Issues

Fixes #317 

### Description of the Change

This PR fixes the inconsistency by allowing TCT to use the same link target as CLM. Before it was like this :

* [CLM](https://github.com/eiffel-community/eiffel/blob/master/eiffel-vocabulary/EiffelConfidenceLevelModifiedEvent.md) allows to [SUBJECT](https://github.com/eiffel-community/eiffel/blob/master/eiffel-vocabulary/EiffelConfidenceLevelModifiedEvent.md#subject) link to CD, ArtC, as well as SCC and SCS 

* [TCT](https://github.com/eiffel-community/eiffel/blob/master/eiffel-vocabulary/EiffelTestCaseTriggeredEvent.md) link [Item Under Test (IUT)](https://github.com/eiffel-community/eiffel/blob/master/eiffel-vocabulary/EiffelTestCaseTriggeredEvent.md#iut) only allows CD or ArtC as targets, enforcing an extra [CD](https://github.com/eiffel-community/eiffel/blob/master/eiffel-vocabulary/EiffelCompositionDefinedEvent.md) with [ELEMENT](https://github.com/eiffel-community/eiffel/blob/master/eiffel-vocabulary/EiffelCompositionDefinedEvent.md#element) link to use SCC as IUT when running tests on source code.

### Alternate Designs

None

### Benefits

Confusing why you can set a confidence level on for example a SCS event but you can't use the same event as a IUT link

### Possible Drawbacks

None that I can think of

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Mattias Linnér mattias.linner@ericsson.com

### CODE REVIEW

**Review by e-backmark-ericsson — APPROVED** [review]

File: definitions/EiffelTestCaseTriggeredEvent/3.2.1.yml [hunk 1]
```diff
@@ -0,0 +1,215 @@
+# Copyright 2017-2022 Ericsson AB and others.
```
Comment (t-persson) [inline definitions/EiffelTestCaseTriggeredEvent/3.2.1.yml:1]:
```suggestion
# Copyright 2017-2023 Ericsson AB and others.
```

File: definitions/EiffelTestCaseTriggeredEvent/3.2.1.yml [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+      primarily to non event-triggered activities. For more information on
+      the usage of this link type please see
+      [Activity Linking](../eiffel-syntax-and-usage/activity-linking.md).'
+    required: false
+    multiple: true
+    targets:
+      any_type: false
+      types:
+        - EiffelTestCaseTriggeredEvent
+_history:
```
Comment (magnusbaeck) [inline definitions/EiffelTestCaseTriggeredEvent/3.2.1.yml:190]: Please add an entry to the history table.

**Review by magnusbaeck — CHANGES_REQUESTED** [review]

**Comment by magnusbaeck** [comment]
Also, I'd kinda expected new target link types to be a minor version bump rather than a patch bump. That's what's proposed in the ArtD PR (#322) where a number of event types can start linking to ArtD.

File: eiffel-vocabulary/EiffelTestCaseTriggeredEvent.md [hunk 3]
```diff
@@ -251,6 +251,7 @@ __Description:__ A URI pointing at a location from where the schema used when cr
 
 | Version | Introduced in | Changes |
 | ------- | ------------- | ------- |
+| 3.3.0 | Current version | Add SCS and SCC as legal target for IUT in TCT (see [Issue 317](https://github.com/eiffel-community/eiffel/issues/317)). |
```
Comment (m-linner-ericsson) [inline eiffel-vocabulary/EiffelTestCaseTriggeredEvent.md:254]: Have any good idea what to write here?

File: definitions/EiffelTestCaseTriggeredEvent/3.2.1.yml (hunk 2 above)
Comment (m-linner-ericsson) [reply definitions/EiffelTestCaseTriggeredEvent/3.2.1.yml]: Fixed in [0b46231](https://github.com/eiffel-community/eiffel/pull/350/commits/0b462317d6c4ba53c29866f24822b6a035450429). However I am not sure about what to write for the "Introduced in" column.

**Comment by m-linner-ericsson** [comment]
> Also, I'd kinda expected new target link types to be a minor version bump rather than a patch bump. That's what's proposed in the ArtD PR (#322) where a number of event types can start linking to ArtD.

@magnusbaeck True, fixed that in [0b46231](https://github.com/eiffel-community/eiffel/pull/350/commits/0b462317d6c4ba53c29866f24822b6a035450429)

File: eiffel-vocabulary/EiffelTestCaseTriggeredEvent.md (hunk 3 above)
Comment (magnusbaeck) [reply eiffel-vocabulary/EiffelTestCaseTriggeredEvent.md:254]: The big table on https://github.com/eiffel-community/eiffel/blob/master/eiffel-syntax-and-usage/event-schemas.md specifies that "No edition set" should be used. That's nonsense, hence #326.

**Review by t-persson — APPROVED** [review]

**Review by e-backmark-ericsson — APPROVED** [review]

**Pull request merged**